pkg/advancedapi/advancedApi_test.go

```diff
@@ -114,59 +117,62 @@ func Test_can_create_new_registered_identity_with_default_issuer_name(t *testing
 		assert.Equal(t, regID.Issuer().Name, c.name)
+		assert.Equal(t, regDoc.ID, regID.Did())
 	}
 }
 
 func Test_can_create_new_registered_identity_will_not_override_doc_if_exists(t *testing.T) {
 	resolver := test.NewInMemoryResolver()
-	regID, err := advancedapi.NewRegisteredIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#ExistingId", false)
+	_, doc, err := advancedapi.RegisterNewIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#ExistingId", false)
 	assert.NilError(t, err)
 
-	doc, err := resolver.GetDocument(regID.Did())
-	assert.NilError(t, err)
 	assert.Check(t, len(doc.PublicKeys) == 1)
 	assert.Check(t, doc.PublicKeys[0].ID == "#ExistingId")
 
-	regID, err = advancedapi.NewRegisteredIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#NewId", false)
+	_, doc, err = advancedapi.RegisterNewIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#NewId", false)
 	assert.NilError(t, err)
 
-	doc, err = resolver.GetDocument(regID.Did())
-	assert.NilError(t, err)
 	assert.Check(t, len(doc.PublicKeys) == 1)
 	assert.Check(t, doc.PublicKeys[0].ID == "#ExistingId")
 }
 
 func Test_can_create_new_registered_identity_will_override_doc_if_exists_and_override_true(t *testing.T) {
 	resolver := test.NewInMemoryResolver()
-	regID, err := advancedapi.NewRegisteredIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#ExistingId", false)
+	_, doc, err := advancedapi.RegisterNewIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#ExistingId", false)
 	assert.NilError(t, err)
 
-	doc, err := resolver.GetDocument(regID.Did())
-	assert.NilError(t, err)
 	assert.Check(t, len(doc.PublicKeys) == 1)
 	assert.Check(t, doc.PublicKeys[0].ID == "#ExistingId")
 
-	regID, err = advancedapi.NewRegisteredIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#NewId", true)
+	_, doc, err = advancedapi.RegisterNewIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#NewId", true)
 	assert.NilError(t, err)
 
-	doc, err = resolver.GetDocument(regID.Did())
-	assert.NilError(t, err)
 	assert.Check(t, len(doc.PublicKeys) == 1)
 	assert.Check(t, doc.PublicKeys[0].ID == "#NewId")
 }
 
 func Test_can_delegate_authentication(t *testing.T) {
 	resolver := test.NewInMemoryResolver()
-	userID, err := advancedapi.NewRegisteredIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#ExistingId", false)
+	userID, userDoc, err := advancedapi.RegisterNewIdentity(resolver, identity.User, test.ValidKeyPairPlop, "#ExistingId", false)
 	assert.NilError(t, err)
-	agentID, err := advancedapi.NewRegisteredIdentity(resolver, identity.User, test.ValidKeyPairPlop2, "#ExistingId", false)
+	agentID, agentDoc, err := advancedapi.RegisterNewIdentity(resolver, identity.User, test.ValidKeyPairPlop2, "#ExistingId", false)
 	assert.NilError(t, err)
 
-	err = advancedapi.DelegateAuthentication(resolver, userID.KeyPair(), userID.Did(), agentID.KeyPair(), agentID.Did(), test.DelegationName)
+	opts := advancedapi.DelegationOpts{
```

Comment:
As far as I can see, this test (and `Test_can_delegate_control`) are the only ones that use the new `DelegatingDocument`/`SubjectDocument` fields. Am I right in thinking there is no easy to test whether we actually avoid the calls? Actually - how about adding counters for register & discover  in the memory resolver - then can assert that not too many calls have been made here.

Reply:
I've added a basic test for this on the highlevel api.

Reply:
Good stuff :+1: